### ISSUE TITLE
Storage ESXI disks not correctly parsed

### ISSUE THREAD

**Issue by puppetninja** [canonical/packer-maas#12]
Hi, 
I am new to ESXI and MAAS

I tried with one disk node and found storage is not correctly configured due to below lines
https://github.com/canonical/packer-maas/blob/master/vmware-esxi/storage-esxi#L74-L77

I see below configs in `curtin.cfg`
```
  - {grub_device: true, id: sda, model: ST1000NM0055-1V410C, name: sda, ptable: gpt,                               
    serial: ZBS181Y7, type: disk, wipe: superblock}  
```
meanwhile 
```
Device                Physical Blocksize  Logical Blocksize  Logical Block Count        Size  Format Type
--------------------  ------------------  -----------------  -------------------  ----------  -----------
naa.5000c500b0009813                 512                512           1953525168  953869 MiB  512n
```

so the disk dict is empty, looks like what being checked is not ESXI disk's pattern ?

I built the image with ESXI 6.7U3b

Thanks

**Comment by ltrager** [comment]
[ESXi has a specific layout you must use.](https://git.launchpad.net/maas/tree/src/maasserver/storage_layouts.py#n775) The VMFS partition, which contains ISOs and OS images, is partition 3 which is stored at the end of the disk. By default this takes up the remaining space on the disk. MAAS allows you to configured multiple VMFS partitions on multiple disks if you like but the core 7 partitions are static.

If you don't specify an ESXi storage config in MAAS the storage config sent is the basic one you posted above. This simply tells Curtin which disk ESXi should be installed on. storage-esxi knows that this means do the default layout which is the 7 base partitions with the 8th taking up the remainder of the disk.

What is the layout are you trying to do? Did you start by selecting the VMware ESXi 6 layout?

**Comment by puppetninja** [comment]
Hi @ltrager 

yes I specify the VMFS6 disk layout in MAAS for storage
so as this is storage part the curtin.cfg
```
storage:                                                                                                                                                                                                                                                                                                     
  config:                                                                                                                                                                                                                                                                                                    
  - {grub_device: true, id: sda, model: ST1000NM0055-1V410C, name: sda, ptable: gpt,                                                                                                                                                                                                                         
    serial: ZBS181Y7, type: disk, wipe: superblock}                                                                                                                                                                                                                                                          
  - {device: sda, flag: boot, id: sda-part1, name: sda-part1, number: 1, size: 3145728B,                                                                                                                                                                                                                     
    type: partition, uuid: null, wipe: superblock}                                                                                                                                                                                                                                                           
  - {device: sda, id: sda-part2, name: sda-part2, number: 2, size: 4294967296B, type: partition,                                                                                                                                                                                                             
    uuid: null, wipe: superblock}                                                                                                                                                                                                                                                                            
  - {device: sda, id: sda-part3, name: sda-part3, number: 3, size: 992280051712B,                                                                                                                                                                                                                            
    type: partition, uuid: 64347da0-8758-4670-94ea-f157bb6ed997, wipe: superblock}                                                                                                                                                                                                                           
  - devices: [sda-part3]                                                                                                                                                                                                                                                                                     
    id: datastore1                                                                                                                                                                                                                                                                                           
    name: datastore1                                                                                                                                                                                                                                                                                         
    type: vmfs6                                                                                                                                                                                                                                                                                              
  - {device: sda, id: sda-part5, name: sda-part5, number: 5, size: 261095424B, type: partition,                                                                                                                                                                                                              
    uuid: null, wipe: superblock}                                                                                                                                                                                                                                                                            
  - {device: sda, id: sda-part6, name: sda-part6, number: 6, size: 261095424B, type: partition,                                                                                                                                                                                                              
    uuid: null, wipe: superblock}                                                                                                                                                                                                                                                                            
  - {device: sda, id: sda-part7, name: sda-part7, number: 7, size: 114294784B, type: partition,                                                                                                                                                                                                              
    uuid: null, wipe: superblock}                                                                                                                                                                                                                                                                            
  - {device: sda, id: sda-part8, name: sda-part8, number: 8, size: 298844160B, type: partition,                                                                                                                                                                                                              
    uuid: null, wipe: superblock}                                                                                                                                                                                                                                                                            
  - {device: sda, id: sda-part9, name: sda-part9, number: 9, size: 2684354560B, type: partition,                                                                                                                                                                                                             
    uuid: null, wipe: superblock}                                                                                                                                                                                                                                                                            
  - {fstype: vmfs6, id: datastore1_format, label: null, type: format, uuid: c1837b44-bfb3-43e9-ba74-f2f122f6957a,                                                                                                                                                                                            
    volume: datastore1}                                                                                                                                                                                                                                                                                      
  - {device: datastore1_format, id: datastore1_mount, path: /vmfs/volumes/datastore1,                                                                                                                                                                                                                        
    type: mount}                                                                                                                                                                                                                                                                                             
  version: 1 
```
what I found is due the check of this line
```
if m or device[0].endswith(serial):
```
after this loop
```
            for device in devices:                              
                # Search for both the model and serial         
                m = re.search('.*%s.*%s.*' % (model, serial), device[0])
                # VMware sometimes only uses the serial in the device name.
                if m or device[0].endswith(serial) or len(devices) == 1:   
                    i['path'] = '/vmfs/devices/disks/%s' % device[0]        
                    i['blocksize'] = int(device[1])                        
                    disks[i['id']] = i                                         
                    break
 ```
`disks` remains to be an empty dict

For now I add a hack like following 
```
if m or device[0].endswith(serial) or len(devices) == 1:
```
instead of
```
if m or device[0].endswith(serial):
```
In this repo

**Comment by ltrager** [comment]
Sorry about that. The primary test environment I have been using always showed disks in the form of model.serial. I just tested on another machine and saw this issue. I believe it is fixed in bbfaf335d960d9ffcfaab9d7dd09a362c89c5a1d. Feel free to reopen if you are still having issues.
